brave/src/main/java/brave/propagation/ThreadLocalCurrentTraceContext.java

```diff
@@ -92,12 +92,21 @@ public void clear() {
   @Override public Scope newScope(@Nullable TraceContext currentSpan) {
     final TraceContext previous = local.get();
     local.set(currentSpan);
-    class ThreadLocalScope implements Scope {
-      @Override public void close() {
-        local.set(previous);
-      }
-    }
-    Scope result = new ThreadLocalScope();
+    Scope result = new ThreadLocalScope(local, previous);
     return decorateScope(currentSpan, result);
   }
+
+  static final class ThreadLocalScope implements Scope {
```

Comment:
note this is not really much different than before.. I can revert it.. I was just playing around.